django-backend/fecfiler/scha_transactions/migrations/0033_memo_description_to_memo_record.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 3.2.12 on 2022-10-31 14:44
+
+from django.db import migrations
+
+
+def text_field_to_text_record(apps, _):
+    sch_a_transaction = apps.get_model("scha_transactions", "schatransaction")  # noqa
+    memo_text = apps.get_model("memo_text", "memotext")
+    for transaction in sch_a_transaction.objects.all():
+        transaction.memo_text = memo_text()
+        transaction.memo_text.text4000 = transaction.memo_text_description
+        transaction.memo_text.transaction_id_number = transaction.id
```

Comment:
In this case the `transaction_id_number`... you're going to love this... is the "transaction" id of the memo, as in a unique id for the memo record, whereas the `back_reference_tran_id_number` is the (FEC, not internal fecfile) id of the transaction it's tied to.  so you want to assign `transaction.memo_text.back_reference_tran_id_number = transaction.transaction_id` and `transaction.memo_text.transaction_id_number = <need to look up convention for this>`

Reply:
Plus  `transaction.memo_text.back_reference_sched_form_name = transaction.form_type`

Reply:
Do we want to save any of that, or do we want to retrieve it from the related transaction?  I'm pretty against saving things twice wherever possible

Reply:
me too.  you could leave the backreference fields till the .fec serialization, but transaction_id_number is information that only pertains to the memo_text record and can't be extrapolated from the transaction it refers to

Reply:
Hmm...  That's a really good point.  It would naturally be a value derived from our UUID's for consistency's sake.  I kind of think we should define a method in the transaction model that gets the FEC-suitable id from the UUID?  Maybe?  Idk...

Reply:
I think that's a good idea, all that matters is that it's unique within the report (across amendments).  like the memo's uuid, but truncated

Reply:
This is already being done, thankfully.  On generation, each transaction creates a transaction_id, which is a 20 character string derived from a UUID that is checked for uniqueness against all other transactions.  We use that for the FEC suitable id that is shown on the front-end, and there's no reason not to use it here.

Reply:
for transactions yeah, but the `transaction_id_number` is called what it's called because the .fec spec didn't change the name of the field for the memo spec.  If I were to name it, I'd call it something like "memo_id".